src/AddressId.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.7;
+
+import {DripsReceiver, ERC20DripsHub, SplitsReceiver} from "./ERC20DripsHub.sol";
+import {IERC20} from "openzeppelin-contracts/token/ERC20/IERC20.sol";
+
+contract AddressId {
+    ERC20DripsHub public immutable dripsHub;
+    uint32 public immutable account;
+
+    /// @param _dripsHub The drips hub to use
+    constructor(ERC20DripsHub _dripsHub) {
+        dripsHub = _dripsHub;
+        account = _dripsHub.createAccount(address(this));
+    }
+
+    /// @notice Calculates the user ID for an address
+    /// @param userAddr The user address
+    /// @return userId The user ID
+    function calcUserId(address userAddr) public view returns (uint256 userId) {
+        return (uint256(account) << 224) | uint160(userAddr);
```

Comment:
If we enforce `require(account == 0)` in the constructor, we could save the bit-shifting.

Reply:
That's true, but it would also create a hard dependency between this contract and a deployment script. The gas savings are nonexistent, it looks like it would shave off a single sequence of `PUSH PUSH SHL OR` per call, 12 gas. The only reason to do that would be to make code more readable, but IMO this explicit approach is better because it's clearly in line with the `userId` model.

Reply:
Okay, make sense then lets keep it.